api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/DurationUnitTranslatorImpl.java

```diff
@@ -0,0 +1,36 @@
+package org.openmrs.module.fhir2.api.translators.impl;
+
+import org.hl7.fhir.r4.model.Timing;
+import org.openmrs.DrugOrder;
+import org.openmrs.module.fhir2.api.translators.DurationUnitTranslator;
+
+import javax.annotation.Nonnull;
+
+public class DurationUnitTranslatorImpl implements DurationUnitTranslator {
+
+	@Override
+	public Timing.UnitsOfTime toFhirResource(@Nonnull DrugOrder drugOrder) {
+
+		String unitUUID = drugOrder.getDurationUnits().getUuid();
+
+		switch (unitUUID) {
+
+			case "162583":
```

Comment:
So these aren't actually what the UUIDs for these concepts would look like in practice. If you look at the concepts as displayed on openconceptlab.org, you'll see an "External ID". That (at least for CIEL) corresponds to the UUID as it would exist in the OpenMRS concept dictionary. For example, [CIEL:1822 (Hour)](https://openconceptlab.org/orgs/CIEL/sources/CIEL/concepts/1822/) has an external id of "1822AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA" (this is actually the pattern for CIEL concept UUIDs: the concept id padded with As to reach 36 characters).

That said, I don't want to have concept ids hard-coded in the translator like this. I'd prefer if we did something closer to what we have for `ObservationCategoryMap` which maps between the [FHIR Observation Category terminology set](https://www.hl7.org/fhir/codesystem-observation-category.html) and OpenMRS concept classes. This support requires a few different components:

1. We have a translator, `ObservationCategoryTranslator` and a corresponding [implementation](https://github.com/openmrs/openmrs-module-fhir2/blob/master/api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/ObservationCategoryTranslatorImpl.java) which adds the observation category to the FHIR resource.

2. We have a simple table that stores mappings between OpenMRS concept classes and the FHIR Observation Category. This table is defined by [this class](https://github.com/openmrs/openmrs-module-fhir2/blob/master/api/src/main/java/org/openmrs/module/fhir2/model/FhirObservationCategoryMap.java).

3. Finally, we have [this simple class](https://github.com/openmrs/openmrs-module-fhir2/blob/master/api/src/main/java/org/openmrs/module/fhir2/api/mappings/ObservationCategoryMap.java) which defines the lookups to get the category given the concept class UUID and to get the concept class UUID using the category.

The flexibility afforded by this model is that we're able to map arbitrary OpenMRS concept dictionaries (not all of which use the CIEL concepts) to the corresponding FHIR terminology. Does that make some kind of sense?